### PR TITLE
Change to reject varag methods

### PR DESCRIPTION
**Pull request by gkhanna79**
@jkotas PTAL and share feedback.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
#1440 lists a bunch more tests with varargs, if we're really set on deleting them.

**Comment by gkhanna79** [comment]
@AndyAyersMS Yes, I am going clean them all up.

File: src/vm/jitinterface.cpp [hunk 1]
```diff
@@ -538,6 +538,15 @@ CEEInfo::ConvToJitSig(
         ULONG data;
         IfFailThrow(sig.GetCallingConvInfo(&data));
         sigRet->callConv = (CorInfoCallConv) data;
+
+#if defined(FEATURE_CORECLR)
+        if (sigRet->callConv & IMAGE_CEE_CS_CALLCONV_VARARG)
+        {
+            // This signature corresponds to a method that uses varargs, which are not supported.
+             COMPlusThrow(kInvalidOperationException, IDS_EE_VARARG_NOT_SUPPORTED);
```
Comment (jkotas) [inline src/vm/jitinterface.cpp]: I think InvalidProgramException would be better.

Comment (jkotas) [inline src/vm/jitinterface.cpp]: Or PlatformNotSupportedException

**Comment by jkotas** [comment]
LGTM modulo minor comment

**Comment by gkhanna79** [comment]
Fixes https://github.com/dotnet/coreclr/issues/4940.

@RussKeldorph @yizhang82 I have deleted tests using varargs that were straightforward. The remaining, I have baselined. Please triage to either delete them or refactor if they contain more than one test within the assembly.

**Comment by yizhang82** [comment]
@gkhanna79 How did you determine those interop tests have varags? I don't believe we have any. I checked one of them (UTF8) and it does not use vararg.

**Comment by gkhanna79** [comment]
The Pinvoke generates it. I ran the entire P1 repo tests against the change and the signature of methods were determined to be varargs.

File: src/vm/jitinterface.cpp [hunk 2]
```diff
@@ -538,6 +538,15 @@ CEEInfo::ConvToJitSig(
         ULONG data;
         IfFailThrow(sig.GetCallingConvInfo(&data));
         sigRet->callConv = (CorInfoCallConv) data;
+
+#if defined(FEATURE_CORECLR)
+        if (sigRet->callConv & IMAGE_CEE_CS_CALLCONV_VARARG)
```
Comment (yizhang82) [inline src/vm/jitinterface.cpp]: This looks wrong. CALLCONV are enum values and not bit flags. 

    IMAGE_CEE_CS_CALLCONV_DEFAULT       = 0x0,

    IMAGE_CEE_CS_CALLCONV_VARARG        = 0x5,
    IMAGE_CEE_CS_CALLCONV_FIELD         = 0x6,
    IMAGE_CEE_CS_CALLCONV_LOCAL_SIG     = 0x7,
    IMAGE_CEE_CS_CALLCONV_PROPERTY      = 0x8,
    IMAGE_CEE_CS_CALLCONV_UNMGD         = 0x9,
    IMAGE_CEE_CS_CALLCONV_GENERICINST   = 0xa,  // generic method instantiation
    IMAGE_CEE_CS_CALLCONV_NATIVEVARARG  = 0xb,  // used ONLY for 64bit vararg PInvoke calls
    IMAGE_CEE_CS_CALLCONV_MAX           = 0xc,  // first invalid calling convention

**Comment by yizhang82** [comment]
@gkhanna79 I believe your change is not correct. CALLCONV are enum values and not bit flags. You should not use & operator.
    IMAGE_CEE_CS_CALLCONV_DEFAULT       = 0x0,
    IMAGE_CEE_CS_CALLCONV_VARARG        = 0x5,
    IMAGE_CEE_CS_CALLCONV_FIELD         = 0x6,
    IMAGE_CEE_CS_CALLCONV_LOCAL_SIG     = 0x7,
    IMAGE_CEE_CS_CALLCONV_PROPERTY      = 0x8,
    IMAGE_CEE_CS_CALLCONV_UNMGD         = 0x9,
    IMAGE_CEE_CS_CALLCONV_GENERICINST   = 0xa,  // generic method instantiation
    IMAGE_CEE_CS_CALLCONV_NATIVEVARARG  = 0xb,  // used ONLY for 64bit vararg PInvoke calls
    IMAGE_CEE_CS_CALLCONV_MAX           = 0xc,  // first invalid calling convention

/cc @jkotas

File: src/vm/jitinterface.cpp (hunk 2 above)
Comment (jkotas) [inline src/vm/jitinterface.cpp]: Yes, you are right. It needs to be: `if (isCallConv(sigRet->callConv, IMAGE_CEE_CS_CALLCONV_VARARG))`

**Comment by gkhanna79** [comment]
I think you are right - part of this enum is used as bit mask and part is not (see GENERIC - it is 0x10). I will revert this (https://github.com/dotnet/coreclr/pull/5121) and get the right fix in.

**Comment by yizhang82** [comment]
@gkhanna79 Yep. You need to use the CORINFO_CALLCONV_MASK. Thanks for quick turn around! :)

File: src/vm/jitinterface.cpp (hunk 2 above)
Comment (gkhanna79) [inline src/vm/jitinterface.cpp]: @jkotas Do we need to check for IMAGE_CEE_CS_CALLCONV_NATIVEVARARG as well?

Comment (jkotas) [inline src/vm/jitinterface.cpp]: It won't hurt to add it as well.

Comment (gkhanna79) [inline src/vm/jitinterface.cpp]: @jkotas The comment says it is used for 64bit PInvokes only but I dont see it being documented or used in that manner. Is it really only used for 64bit PInvoke?

Comment (jkotas) [inline src/vm/jitinterface.cpp]: It is used in PInvoke IL stubs - on all platforms

**Pull request merged**